net/strongswan/files/ipsec.init

```diff
@@ -1,20 +1,327 @@
 #!/bin/sh /etc/rc.common
 
-START=90
-STOP=10
+NAME=ipsec
+START=21
+STOP=60
+USE_PROCD=1
 
-start() {
-	ipsec start
+. $IPKG_INSTROOT/lib/functions.sh
+. $IPKG_INSTROOT/lib/functions/service.sh
+. $IPKG_INSTROOT/lib/functions/network.sh
+
+FileSecrets=/var/ipsec/ipsec.secrets
+FileConn=/var/ipsec/ipsec.conf
+FileCommon=/var/ipsec/strongswan.conf
+
+EnabledRemotePeers=0
+
+ConfigPhase1() {
+	local encryption_algorithm
+	local hash_algorithm
+	local dh_group
+
+	config_get encryption_algorithm  "$1" encryption_algorithm
+	config_get hash_algorithm        "$1" hash_algorithm
+	config_get dh_group              "$1" dh_group
+
+	Phase1Proposal=${Phase1Proposal}","${encryption_algorithm}-${hash_algorithm}-${dh_group}
+}
+
+ConfigConn() {
+	# Generic ipsec conn section shared by tunnel and transport
+	local mode
+	local local_subnet
+	local local_nat
+	local local_sourceip
+	local local_updown
+	local local_firewall
+	local remote_subnet
+	local remote_sourceip
+	local remote_updown
+	local remote_firewall
+	local ikelifetime
+	local lifetime
+	local margintime
+	local keyingtries
+	local dpdaction
+	local dpddelay
+	local inactivity
+	local keyexchange
+
+	config_get mode                     "$1"           mode "route"
+	config_get local_subnet             "$1"           local_subnet
+	config_get local_nat                "$1"           local_nat ""
+	config_get local_sourceip           "$1"           local_sourceip ""
+	config_get local_updown             "$1"           local_updown ""
+	config_get local_firewall           "$1"           local_firewall ""
+	config_get remote_subnet            "$1"           remote_subnet
+	config_get remote_sourceip          "$1"           remote_sourceip ""
+	config_get remote_updown            "$1"           remote_updown ""
+	config_get remote_firewall          "$1"           remote_firewall ""
+	config_get p2_proposal              "$1"           p2_proposal
+	config_get ikelifetime              "$1"           ikelifetime "3h"
+	config_get lifetime                 "$1"           lifetime "1h"
+	config_get margintime               "$1"           margintime "9m"
+	config_get keyingtries              "$1"           keyingtries "3"
+	config_get dpdaction                "$1"           dpdaction "none"
+	config_get dpddelay                 "$1"           dpddelay "30s"
+	config_get inactivity               "$1"           inactivity
+	config_get keyexchange              "$1"           keyexchange "ikev2"
+
+	[[ "$local_nat" != "" ]] && local_subnet=$local_nat
+
+	echo "conn $ConfigName-$1" >> $FileConn
+	echo "  left=%any" >> $FileConn
+	echo "  right=$RemoteGateway" >> $FileConn
```

Comment:
Please use a helper function instead of echo directly, like xappend in dnsmasq init script.